### PR TITLE
[Xamarin.Android.Build.Tasks] Add $(UseInterpreter)

### PR DESCRIPTION
**Pull request by pjcollins**
Fixes: https://github.com/xamarin/xamarin-android/issues/5824
Context: https://github.com/mono/mono/issues/19669

A new `$(UseInterpreter)` property has been added to toggle the existing
`$(AndroidUseInterpreter)` setting.  This new property does not contain
OS / Platform information so that it can be used by multi-targeted
projects.

The `<BuildApk/>` task no longer needs to check the list of supported
ABIs when the interpreter is enabled.  The interpreter should now work
with all architectures.  I ran the .NET 6 version of the new test on an
x86 emulator locally and the app launched successfully.

There was also an oversight in the legacy targets that caused
interpreter builds to fail with the following, which has been fixed:

    (_BuildApkEmbed target) ->
      C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2010,3): error XABLD7028: System.IO.FileNotFoundException: Could not find file 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\lib\interpreter-arm64-v8a\libxa-internal-api.so'.
      C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2010,3): error XABLD7028: File name: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\lib\interpreter-arm64-v8a\libxa-internal-api.so'

### CODE REVIEW

**Review by grendello — APPROVED** [review]

**Review by grendello — APPROVED** [review]

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Legacy.targets [hunk 1]
```diff
@@ -519,8 +519,8 @@ projects. .NET 5 projects will not import this file.
...
       <FrameworkNativeLibrary Include="%(_TargetLibDir.Identity)\libmono-android-checked+$(_AndroidCheckedBuild).release.so" Condition=" '$(AndroidIncludeDebugSymbols)' != 'True' And '$(_AndroidCheckedBuild)' != '' " ArchiveFileName="libmonodroid.so" />
-      <FrameworkNativeLibrary Include="%(_TargetLibInterpreterDir.Identity)\libxa-internal-api.so" Condition=" '$(_AndroidCheckedBuild)' == ''" />
-      <FrameworkNativeLibrary Include="%(_TargetLibInterpreterDir.Identity)\libxa-internal-api-checked+$(_AndroidCheckedBuild).so" Condition=" '$(_AndroidCheckedBuild)' != '' " ArchiveFileName="libxa-internal-api.so" />
+      <FrameworkNativeLibrary Include="%(_TargetLibDir.Identity)\libxa-internal-api.so" Condition=" '$(_AndroidCheckedBuild)' == ''" />
+      <FrameworkNativeLibrary Include="%(_TargetLibDir.Identity)\libxa-internal-api-checked+$(_AndroidCheckedBuild).so" Condition=" '$(_AndroidCheckedBuild)' != '' " ArchiveFileName="libxa-internal-api.so" />
       <FrameworkNativeLibrary Include="%(_TargetLibInterpreterDir.Identity)\libmono-btls-shared.so" Condition=" '$(_AndroidDebugNativeLibraries)' != 'True' " />
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Legacy.targets:524]: What is the value of `@(_TargetLibDir)` vs `@(_TargetLibInterpreterDir)` here? Were some of these wrong and we didn't notice?

Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Legacy.targets:524]: I think this was just an error in the initial PR, but maybe @grendello can help confirm.  Should there be copies of these `libxa-internal-api` files in the interpreter folders in our installers?

When I added the interpreter test initially I hit the following:

```
(_BuildApkEmbed target) ->
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2010,3): error XABLD7028: System.IO.FileNotFoundException: Could not find file 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\lib\interpreter-arm64-v8a\libxa-internal-api.so'.
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2010,3): error XABLD7028: File name: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\lib\interpreter-arm64-v8a\libxa-internal-api.so'
```

The interpreter lib folders only seem to have mono items in them:

```
    Directory: C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\lib\interpreter-arm64-v8a

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---            3/5/2021 11:02 AM        1613872 libmono-btls-shared.so
-a---            3/5/2021 11:02 AM         728184 libmono-native.so
-a---            3/5/2021 11:02 AM         301904 libMonoPosixHelper.so
-a---            3/5/2021 11:02 AM         173928 libmono-profiler-aot.so
-a---            3/5/2021 11:02 AM         284472 libmono-profiler-log.so
-a---            3/5/2021 11:02 AM        4285104 libmonosgen-2.0.so
```

These `libxa-internal-api` files only exist in the regular ABI folders:

```
    Directory: C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\lib\arm64-v8a

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---            3/5/2021 11:02 AM        4602235 installer
-a---            3/5/2021 11:02 AM         305640 libmono-android.debug.so
-a---            3/5/2021 11:02 AM         263408 libmono-android.release.so
-a---            3/5/2021 11:02 AM         799728 libmono-android-checked+asan.debug.so
-a---            3/5/2021 11:02 AM         644800 libmono-android-checked+asan.release.so
-a---            3/5/2021 11:02 AM        1277072 libmono-android-checked+ubsan.debug.so
-a---            3/5/2021 11:02 AM        1153440 libmono-android-checked+ubsan.release.so
-a---            3/5/2021 11:02 AM        6174856 libmono-btls-shared.d.so
-a---            3/5/2021 11:02 AM        1613872 libmono-btls-shared.so
-a---            3/5/2021 11:02 AM        2201704 libmono-native.d.so
-a---            3/5/2021 11:02 AM         707024 libmono-native.so
-a---            3/5/2021 11:02 AM         995832 libMonoPosixHelper.d.so
-a---            3/5/2021 11:02 AM         301904 libMonoPosixHelper.so
-a---            3/5/2021 11:02 AM         173928 libmono-profiler-aot.so
-a---            3/5/2021 11:02 AM         284472 libmono-profiler-log.so
-a---            3/5/2021 11:02 AM       18787048 libmonosgen-2.0.d.so
-a---            3/5/2021 11:02 AM        4033248 libmonosgen-2.0.so
-a---            3/5/2021 11:02 AM        1191016 libsqlite3_xamarin.so
-a---            3/5/2021 11:02 AM          65280 libxa-internal-api.so
-a---            3/5/2021 11:02 AM          91728 libxa-internal-api-checked+asan.so
-a---            3/5/2021 11:02 AM         585048 libxa-internal-api-checked+ubsan.so
-a---            3/5/2021 11:02 AM          37256 libxamarin-debug-app-helper.so
-a---            3/5/2021 11:02 AM          84216 libxamarin-debug-app-helper-checked+asan.so
-a---            3/5/2021 11:02 AM         370040 libxamarin-debug-app-helper-checked+ubsan.so
-a---            3/5/2021 11:02 AM         172664 xamarin.find
-a---            3/5/2021 11:02 AM           5280 xamarin.stat
-a---            3/5/2021 11:02 AM         174600 xamarin.sync
```

Comment (grendello) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Legacy.targets:524]: It was an error, yes. The internal API library is the same no matter whether interpreter or standard Mono runtime is used.

**Review by dellis1972 — APPROVED** [review]

**Review by jonathanpeppers — APPROVED** [review]
I restarted a couple tests, `ClassLibraryMainLauncherRuns` should be fixed in 522d7fb61f3669d85d077ba4f2889dbe8c9c8ac3.

**Pull request merged**